### DISCUSSION TITLE
Plans to support WebView.InvokeScript?

### DISCUSSION THREAD

**Discussion by nikolayvpavlov** [unoplatform/uno#10251] [Q&A] [answered]
Are there plans to add support for WebView.InvokeScript?

I need to develop a Windows (WinUI) and iOS application with a map control.  Since WinUI does not have a map control, Uno's MapControl is not supported on Windows.  The alternative is to use Azure MapControl in an embedded WebView.  However, in order to interact with it, I need to be able to call JS functions.

This is a show-stopper for my project and will make me use MAUI.

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the feedback. Which target platforms are you looking at?

**Reply by jeromelaban** [reply]
I missed the iOS part. [InvokeScript is supported](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Controls/WebView/WebView.iOSmacOS.cs#L75-L76), so can you explain makes you say that it's not supported?

**Reply by nikolayvpavlov** [reply]
For the first release, I must target Windows (WinUI 3) and iOS (iPhone).

InvokeScript is not supported on Windows (WinUI), according to the documentation. Can't find support for it in the source code either.

**Reply by jeromelaban** [reply]
On Windows, [WebView2.ExecuteScriptAsync](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.wpf.webview2.executescriptasync?view=webview2-dotnet-1.0.705.50#microsoft-web-webview2-wpf-webview2-executescriptasync(system-string)) is available. Where did you find the references to `InvokeScript` for WinUI?

**Reply by nikolayvpavlov** [reply]
The [documentation ](https://platform.uno/docs/articles/implemented/windows-ui-xaml-controls-webview.html)does not mention WebView2 at all.  What is more, it states:

![image](https://user-images.githubusercontent.com/32266447/197808005-eec41ed6-a0ce-456e-bede-a2f92009bb49.png)

And indeed I did not find references to InvokeScript for WinUI.  

Anyway, I understand I must use platform-specific Xaml / C# and use WebView2 for the WinUI target.  

Thank you for your assistance, but I believe the documentation for WebView should be updated.

**Reply by jeromelaban** [reply]
I see, we implemented `InvokeScriptAsync`, not `InvokeScript` (which is a problematic API to begin with), [the documentation is correct](https://platform.uno/docs/articles/implemented/windows-ui-xaml-controls-webview.html#implemented-methods).

For WebView2, indeed, we have not ported the Uno version of WebView to WebView2 yet (https://github.com/unoplatform/uno/issues/4758), so you'll need so platform-specific code here.

**Reply by nikolayvpavlov** [reply]
Indeed I overlooked `InvokeScriptAsync`, for which I apologize.  Thank you for the clarification on WebView2.
